gobblin-api/src/main/java/gobblin/configuration/State.java

```diff
@@ -98,6 +102,19 @@ public void setProp(String key, Object value) {
   }
 
   /**
+   * Appends the input value to a list property that can be retrieved with {@link #getPropAsList}.
+   * @param key property key
+   * @param value property value (should not include commas, which are used as separators)
```

Comment:
Looking at Hadoop Configuration, which supports property lists as comma separated strings, it suffers from the same issues as the current implementation of State (see https://hadoop.apache.org/docs/current/api/org/apache/hadoop/conf/Configuration.html#setStrings(java.lang.String, java.lang.String...) ). 
Hadoop docs do not go extensively into the comma issue (but I looked at the source code). I think it is generally understood that values containing commas always have the risk of being interpreted as lists, and this is doing the same as a well known project as Hadoop is doing.
I don't think there's a benefit in adding a more complex API, or explaining to much in the doc (although I will add some more explanation). Making the separator configurable just adds more configurations that don't contribute much.